internal/services/appconfiguration/app_configuration_feature_resource.go

```diff
@@ -0,0 +1,428 @@
+package appconfiguration
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/appconfiguration/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/appconfiguration/sdk/1.0/appconfiguration"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/appconfiguration/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+const (
+	FeatureKeyContentType = "application/vnd.microsoft.appconfig.ff+json;charset=utf-8"
+	FeatureKeyPrefix      = ".appconfig.featureflag"
+)
+
+type FeatureResource struct {
+}
+
+var _ sdk.ResourceWithUpdate = FeatureResource{}
+
+type FeatureResourceModel struct {
+	ConfigurationStoreId string                       `tfschema:"configuration_store_id"`
+	Description          string                       `tfschema:"description"`
+	Enabled              bool                         `tfschema:"enabled"`
+	Name                 string                       `tfschema:"name"`
+	Label                string                       `tfschema:"label"`
+	Locked               bool                         `tfschema:"locked"`
+	Tags                 map[string]interface{}       `tfschema:"tags"`
+	PercentageFilters    []PercentageFilterParameters `tfschema:"percentage_filter"`
+	TimewindowFilters    []TimewindowFilterParameters `tfschema:"timewindow_filter"`
+	TargetingFilters     []TargetingFilterAudience    `tfschema:"targeting_filter"`
+}
+
+func (k FeatureResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"configuration_store_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: azure.ValidateResourceID,
+		},
+		"description": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+		"enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+		},
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotWhiteSpace,
+		},
+		"etag": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+			Optional: true,
+		},
+		"label": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ForceNew: true,
+		},
+		"locked": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+		"percentage_filter": {
```

Comment:
would it make sense just to unroll this into
```suggestion
		"percentage_filter_value": {
```
?